File: src/domain/repositories/FolderImageRepository.ts
import type { FolderImage } from '@/domain/entities/FolderImage'
import type { GeoLocation } from '@/domain/value-objects/GeoLocation'

export interface ImageFilePayload {
  fileName: string
  contentType: string
  sizeBytes: number
  data: Blob
}

export interface CreateFolderImageInput {
  folderId: string
  dateId: string
  file: ImageFilePayload
  uploadedById: string
  uploadedByName: string
  location?: GeoLocation
}

export interface FolderImageRepository {
  listByFolder(folderId: string): Promise<FolderImage[]>
  listByFolderIds(folderIds: string[]): Promise<FolderImage[]>
  listByDate(folderId: string, dateId: string): Promise<FolderImage[]>
  create(input: CreateFolderImageInput): Promise<FolderImage>
  delete(image: FolderImage): Promise<void>
  deleteAllByFolder(folderId: string): Promise<void>
  deleteAllByDate(folderId: string, dateId: string): Promise<void>
}

File: consorcio/lib/domain/repositories/folder_image_repository.dart
import 'dart:typed_data';

import '../entities/folder_image.dart';
import '../value_objects/geo_location.dart';

class ImageFilePayload {
  const ImageFilePayload({
    required this.fileName,
    required this.contentType,
    required this.bytes,
  });

  final String fileName;
  final String contentType;
  final Uint8List bytes;

  int get sizeBytes => bytes.length;
}

abstract class FolderImageRepository {
  Future<List<FolderImage>> listByFolder(String folderId);
  Future<List<FolderImage>> listByFolderIds(List<String> folderIds);
  Future<List<FolderImage>> listByDate(String folderId, String dateId);
  Future<FolderImage> create({
    required String folderId,
    required String dateId,
    required ImageFilePayload file,
    required String uploadedById,
    required String uploadedByName,
    GeoLocation? location,
  });
}
